infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationManager.java

```diff
@@ -0,0 +1,301 @@
+                            .tsPasswordFile(corfuRuntime.getParameters().getTsPasswordFile())
+                            .build();
+                    CorfuLogReplicationRuntime replicationRuntime = new CorfuLogReplicationRuntime(parameters, metadataManager);
+                    replicationRuntime.start();
+                    runtimeToRemoteCluster.put(remoteCluster.getClusterId(), replicationRuntime);
+                } catch (Exception e) {
+                    log.error("Exception {}. Failed to connect to remote cluster {}. Retry after 1 second.",
+                            e, remoteCluster.getClusterId());
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unrecoverable exception when attempting to connect to remote cluster.", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Stop Log Replication to a specific standby Cluster
+     */
+    private void stopLogReplicationRuntime(String remoteClusterId) {
+        CorfuLogReplicationRuntime logReplicationRuntime = runtimeToRemoteCluster.get(remoteClusterId);
+        if (logReplicationRuntime != null) {
+            log.info("Stop log replication runtime to remote cluster id={}", remoteClusterId);
+            logReplicationRuntime.stop();
+            runtimeToRemoteCluster.remove(remoteClusterId);
+        } else {
+            log.warn("Runtime not found to remote cluster {}", remoteClusterId);
+        }
+    }
+
+    /**
+     * The notification of change of adding/removing standby's without epoch change.
+     *
+     * @param newConfig has the same topologyConfigId as the current config
+     */
+    public void processStandbyChange(TopologyDescriptor newConfig) {
+        if (newConfig.getTopologyConfigId() != topologyDescriptor.getTopologyConfigId()) {
+            log.error("Detected changes in the topology. The new topology descriptor {} doesn't have the same " +
+                    "siteConfigId as the current one {}", newConfig, topologyDescriptor);
+            return;
+        }
+
+        Map<String, ClusterDescriptor> newStandbys = newConfig.getStandbyClusters();
+        Map<String, ClusterDescriptor> currentStandbys = topologyDescriptor.getStandbyClusters();
+        newStandbys.keySet().retainAll(currentStandbys.keySet());
+        Set<String> standbysToRemove = currentStandbys.keySet();
+        standbysToRemove.removeAll(newStandbys.keySet());
+
+        /*
+         * Remove standbys that are not in the new config
+         */
+        for (String siteID : standbysToRemove) {
+            stopLogReplicationRuntime(siteID);
+            topologyDescriptor.removeStandbySite(siteID);
+        }
+
+        //Start the standbys that are in the new config but not in the current config
+        for (String clusterId : newConfig.getStandbyClusters().keySet()) {
```

Comment:
I am wondering if it works as expected.
Let's say currentStandbys.keySet() = [2, 3, 4] and newStandbys.keySet() = [1, 2, 3]. So 1 got added and 4 got removed. We do retainAll on newStandbys on line 207, this will make it newStandbys.keySet() = [2, 3] and 1 (a new standby that is not in the current config and is only in the new config) will be left out?